pyasn1/codec/ber/decoder.py

```diff
@@ -1648,7 +1690,11 @@ def __call__(self, substrate, asn1Spec=None,
 #:    SequenceOf:
 #:     1 2 3
 #:
-decode = Decoder(tagMap, typeMap)
+def decode(substrate, asn1Spec=None, **kwargs):
+    stream = asStream(substrate)
+    value = decodeStream(stream, asn1Spec, **kwargs)
+    return value, stream.read()
```

Comment:
If we are changing the API, should we just not return the remaining octets?  The caller can always keep reading from the stream from where the decoder left off...

Reply:
I suggest to do that - it's trivial but first I want to have a version that has exactly the same API, only with different implementation. Therefore the decode/decodeStream differentiation.

The many dependent projects (including pyasn1-modules) don't have to change a single line at the moment.

Reply:
Right, having a compatibility wrapper makes perfect sense to me!